components/style/workqueue.rs

```diff
@@ -1,385 +0,0 @@
-    chan: Sender<SupervisorMsg<QueueData, WorkData>>,
-    /// The thief end of the work-stealing deque for all other workers.
-    other_deques: Vec<Stealer<WorkUnit<QueueData, WorkData>>>,
-    /// The random number generator for this worker.
-    rng: XorShiftRng,
-}
-
-unsafe impl<QueueData: 'static, WorkData: 'static + Send> Send for WorkerThread<QueueData, WorkData> {}
-
-const SPINS_UNTIL_BACKOFF: u32 = 128;
-const BACKOFF_INCREMENT_IN_US: u32 = 5;
-const BACKOFFS_UNTIL_CONTROL_CHECK: u32 = 6;
-
-#[cfg(not(windows))]
-fn sleep_microseconds(usec: u32) {
-    unsafe {
-        libc::usleep(usec);
-    }
-}
-
-#[cfg(windows)]
-fn sleep_microseconds(_: u32) {
-    unsafe {
-        kernel32::Sleep(0);
-    }
-}
-
-impl<QueueData: Sync, WorkData: Send> WorkerThread<QueueData, WorkData> {
-    /// The main logic. This function starts up the worker and listens for
-    /// messages.
-    fn start(&mut self) {
-        let deque_index_mask = (self.other_deques.len() as u32).next_power_of_two() - 1;
-        loop {
-            // Wait for a start message.
-            let (mut deque, ref_count, queue_data) = match self.port.recv().unwrap() {
-                WorkerMsg::Start(deque, ref_count, queue_data) => (deque, ref_count, queue_data),
-                WorkerMsg::Stop => panic!("unexpected stop message"),
-                WorkerMsg::Exit => return,
-                WorkerMsg::HeapSizeOfTLS(f) => {
-                    self.chan.send(SupervisorMsg::HeapSizeOfTLS(f())).unwrap();
-                    continue;
-                }
-            };
-
-            let mut back_off_sleep = 0 as u32;
-
-            // We're off!
-            'outer: loop {
-                let work_unit;
-                match deque.pop() {
-                    Some(work) => work_unit = work,
-                    None => {
-                        // Become a thief.
-                        let mut i = 0;
-                        loop {
-                            // Don't just use `rand % len` because that's slow on ARM.
-                            let mut victim;
-                            loop {
-                                victim = self.rng.next_u32() & deque_index_mask;
-                                if (victim as usize) < self.other_deques.len() {
-                                    break
-                                }
-                            }
-
-                            match self.other_deques[victim as usize].steal() {
-                                Empty | Abort => {
-                                    // Continue.
-                                }
-                                Data(work) => {
-                                    work_unit = work;
-                                    back_off_sleep = 0 as u32;
-                                    break
-                                }
-                            }
-
-                            if i > SPINS_UNTIL_BACKOFF {
```

Comment:
After spending a few minutes studying Servo profiler dumps, I can't really draw a strong conclusion about this. Sometimes slower, sometimes faster, but always on the same order...